network-api/networkapi/nav/blocks.py

```diff
@@ -17,14 +19,50 @@ def open_in_new_window(self) -> bool:
         return False
 
 
-class NavLinkBlock(BaseLinkBlock):
+class NavItem(BaseLinkBlock):
     description = blocks.CharBlock(required=False, max_length=100)
 
+    def __init__(self, local_blocks=None, **kwargs):
+        # Use __init__ method to change the order of the blocks when constructing
+        # them through inheritance
+        super().__init__(local_blocks, **kwargs)
+        self.child_blocks = self.base_blocks.copy()
+        child_blocks = OrderedDict(
+            {
+                "label": self.child_blocks.pop("label"),
+                "description": self.child_blocks.pop("description"),
+            }
+        )
+        child_blocks.update({k: v for k, v in self.child_blocks.items()})
+        self.child_blocks = child_blocks
+
     class Meta:
-        value_class = NavLinkValue
+        value_class = NavItemValue
         label = "Navigation Link"
         icon = "link"
         template = "nav/blocks/nav_link_block.html"
 
 
-register(BaseLinkBlockAdapter(), NavLinkBlock)
+register(BaseLinkBlockAdapter(), NavItem)
+
+
+class NavButton(BaseLinkBlock):
+    class Meta:
+        value_class = NavItemValue
+        label = "Navigation Button"
+        icon = "link"
+        template = "nav/blocks/nav_button_block.html"
+
+
+register(BaseLinkBlockAdapter(), NavButton)
+
+
+class NavColumn(blocks.StructBlock):
+    title = blocks.CharBlock(max_length=100)
+    links = blocks.ListBlock(NavItem, min_num=1, max_num=4)
+    button = blocks.ListBlock(NavButton, required=False, min_num=0, max_num=1)
```

Comment:
Hey @jhonatan-lopes, is this dev pattern of encompassing a block w/ max_num=1 in a ListBlock useful for:

1. Cleaner UI by hiding fields if child block not used
2. Additional control by not requiring the child block itself (and then requiring the child block fields if it's been added)
3. Anything else I'm missing?

Reply:
Hey @robdivincenzo, this is documented here: https://docs.wagtail.org/en/stable/topics/streamfield.html#limiting-block-counts

Basically, we can use `min_num` and `max_num` to limit the number of blocks in a `ListBlock` or `StreamBlock` combination. In this case, we need at least one link and at most 4. Button, on the other hand, is optional, so we use the `ListBlock` with `min_num=0` and `max_num=1`.